invenio_rdm_records/services/access/service.py

```diff
@@ -749,3 +754,54 @@ def create_guest_access_request(self, identity, token, expand=False, uow=None):
             data=comment,
             uow=uow,
         )
+
+    def _translate_expiration_date(self, value):
+        """Translate secret_link_expiration from ui dropdown value."""
+        if value == 0:
+            return None
+        else:
+            return value
+
+    @unit_of_work()
+    def update_access_settings(
+        self,
+        identity,
+        id_,
+        data,
+        uow=None,
+    ):
+        """Update access settings for a record (resp. its parent)."""
+        record, parent = self.get_parent_and_record_or_draft(id_)
+
+        # Permissions
+        self.require_permission(identity, "manage", record=record)
+
+        expiration_days = self._translate_expiration_date(
+            data["access"]["settings"]["secret_link_expiration"]
+        )
+        data["access"]["settings"]["secret_link_expiration"] = expiration_days
+
+        # Validation
+        data, __ = self.schema_parent.load(
+            data, context=dict(identity=identity), raise_errors=True
+        )
+
+        # Update
+        settings_dict = data["access"]["settings"]
+        for key in settings_dict:
+            setattr(parent.access.settings, key, settings_dict[key])
+
+        # Commit
+        uow.register(RecordCommitOp(parent))
+        if record:
```

Comment:
is `record` optional? Because I can see that it is checked in the permissions in line `777`.

Reply:
Got this fixed in the merged PR: https://github.com/inveniosoftware/invenio-rdm-records/pull/1385